Source/Core/DiscIO/VolumeDirectory.cpp

```diff
@@ -460,18 +461,18 @@ void CVolumeDirectory::WriteDirectory(const File::FSTEntry& parent_entry, u32* f
     // For some reason, sorting by lowest ASCII value first prevents many games from
     // fully booting. We make the comparison case insensitive to solve the problem.
     // (Highest ASCII value first seems to work regardless of case sensitivity.)
-    const std::string one_lower = ASCIIToLowercase(one.virtualName);
-    const std::string two_lower = ASCIIToLowercase(two.virtualName);
-    return one_lower == two_lower ? one.virtualName < two.virtualName : one_lower < two_lower;
+    const std::string one_upper = ASCIIToUppercase(one.virtualName);
+    const std::string two_upper = ASCIIToUppercase(two.virtualName);
+    return one_upper == two_upper ? one.virtualName < two.virtualName : one_upper < two_upper;
   });
 
   for (const File::FSTEntry& entry : sorted_entries)
   {
     if (entry.isDirectory)
     {
       u32 entry_index = *fst_offset / ENTRY_SIZE;
-      WriteEntryData(fst_offset, DIRECTORY_ENTRY, *name_offset, parent_entry_index,
-                     entry_index + entry.size + 1);
+      WriteEntryData(fst_offset, DIRECTORY_ENTRY, *name_offset,
+                     parent_entry_index << m_address_shift, entry_index + entry.size + 1);
```

Comment:
It's a bit hacky that we're adding this to cancel out the `data_offset >> m_address_shift` that happens inside WriteEntryData. I would prefer having an `address_shift` parameter for WriteEntryData instead, so we could use `address_shift = 0` when writing directories and `address_shift = m_address_shift` when writing files.